### PR TITLE
IDE changes for analyzer loading

### PR DESCRIPTION
**Pull request by jaredpar**
For #69406 I updated several places where the IDE created analyzers. Even though I specifically called out the PR to the IDE team, I forgot I'd changed their code and merged once I had two compiler sign offs. This PR is commenting the parts of the IDE I changed in the PR so I can get actual sign off on these changes.

### CODE REVIEW

File: src/Tools/ExternalAccess/OmniSharp/Analyzers/OmnisharpAnalyzerLoaderFactory.cs
Comment (jasonmalinowski) [inline src/Tools/ExternalAccess/OmniSharp/Analyzers/OmnisharpAnalyzerLoaderFactory.cs:14]: Should be fine. It looks like we never actually passed a baseDirectory in the first place. My only concern would be if the name is too long and we get closer to MAX_PATH, but we'll get feedback easily enough if that's the case.

File: src/Workspaces/Core/Portable/Diagnostics/DefaultAnalyzerAssemblyLoaderService.cs
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Diagnostics/DefaultAnalyzerAssemblyLoaderService.cs:17]: Still good, no concerns.

File: src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoader.cs
Comment (jasonmalinowski) [inline src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoader.cs:21]: I left a comment in the original PR but the tl;dr is this should be fine.

**Review by jasonmalinowski — COMMENTED** [review]
I admit I'm not sure which GitHub radio button to choose because:

1. I believe your code is correct, but...
2. Please don't merge this PR with these comments.

So I sign off, but without pressing the signoff button.

**Comment by jaredpar** [comment]
@RikkiGibson, @cston, @jjonescz  PTAL. Small changes from previous PR based on IDE feedback. The diff isn't rendering the best. Suggest looking at the current state of the file. It's a fairly simple change, just diff isn't rendering well.

**Review by cston — APPROVED** [review]

**Review by RikkiGibson — APPROVED** [review]

**Review by jjonescz — APPROVED** [review]

**Pull request merged**
